app/models/reports/repository.rb

```diff
@@ -106,6 +110,42 @@ def initialize(regions, periods:)
       end
     end
 
+    memoize def registration_counts_by_user
+      region_entries = regions.map { |region| RegionEntry.new(region, __method__, period_type: period_type) }
+      region_entries.each_with_object({}) do |region_entry, sum|
+        sum[region_entry.slug] = RegisteredPatientsQuery.new.count(region_entry.region, period_type, group_by: :registration_user_id)
+      end
+    end
+
+    # Returns counts of the cumulative Registration counts done by Users within a Region.
+    # Returns a nested Hash structure in the following shape:
+    # {
+    #   slug => {
+    # .   period => {
+    # .     user_id_1 => cumulative_count,
+    # .     user_id_2 => cumulative_count,
+    # .  }
+    # }
+    #
+    memoize def cumulative_registration_counts_by_user
+      registration_counts_by_user.each_with_object({}) do |(slug, period_counts), totals|
+        totals[slug] = {}
+        # collect all the user ids in a region that we need to count for
+        user_ids = period_counts.each_with_object(Set.new) { |(period, counts), sum| sum.merge(counts.keys) }
```

Comment:
Can we rename:
- `counts` to `user_counts`
- `sum` to `user_ids`

Reply:
Can this be simplified to something like `user_ids = period_counts.values.map(&:keys).flatten.to_set`